### PR TITLE
Metadata Listing UI

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds the metadata listing on the sample#show view. The view now has tabs for attachments and metadata listings.

Currently, metadata filtering, sorting, and pagination will not be included.

## Screenshots or screen recordings
image
image

## How to set up and validate locally
1. Add metadata to a sample
2. Navigate to the sample page
3. As `turbo_stream` was modified for much of the page actions, test all page actions and ensure proper redirects and re-renders.
- Upload files
- Concatenate files
- Delete files
- Edit sample
- Remove sample
4. Ensure clicking between the files and metadata tabs list the correct info in the table below
5. Ensure the sample metadata is accurately displayed.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/views/projects/samples/metadata/_table.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    class="w-full text-sm text-left text-slate-500 dark:text-slate-400 mb-4"
+    data-controller="selection"
+    data-selection-action-link-outlet=".action-link"
+    id="metadata-table"
+  >
+    <thead class="text-slate-700 bg-slate-50 dark:bg-slate-900 dark:text-slate-400">
+      <tr>
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.metadata_field") %></th>
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.value") %></th>
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.provider") %></th>
```
Comment (deepsidhu85) [inline app/views/projects/samples/metadata/_table.html.erb:12]: Can you update the Provider header to `Source` instead

File: app/views/projects/samples/metadata/_table.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<div class="table-container">
+  <table
+    class="w-full text-sm text-left text-slate-500 dark:text-slate-400 mb-4"
+    data-controller="selection"
+    data-selection-action-link-outlet=".action-link"
+    id="metadata-table"
+  >
+    <thead class="text-slate-700 bg-slate-50 dark:bg-slate-900 dark:text-slate-400">
+      <tr>
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.metadata_field") %></th>
```
Comment (deepsidhu85) [inline app/views/projects/samples/metadata/_table.html.erb:10]: Let's update `Metadata Field` header to `Key`

File: app/views/projects/samples/metadata/_table.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.value") %></th>
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.provider") %></th>
+        <th scope="col" class="px-6 py-3"><%= t("projects.samples.show.table_header.last_modified") %></th>
```
Comment (deepsidhu85) [inline app/views/projects/samples/metadata/_table.html.erb:13]: Let's update `Last Modified` header to `Last Updated`

File: app/controllers/projects/samples_controller.rb [hunk 4]
```diff
@@ -40,7 +46,9 @@ def create
 
       if @sample.persisted?
         flash[:success] = t('.success')
-        redirect_to namespace_project_sample_path(id: @sample.id)
+        respond_to do |format|
```
Comment (deepsidhu85) [inline app/controllers/projects/samples_controller.rb:49]: You can revert this change and add a data: {turbo: false} to the form signature in `samples/new.html.erb`. It will look like this:

`<%= form_with(model: @sample, url: namespace_project_samples_path, data: {turbo: false}, method: :post) do |form| %>`

This will properly redirect from the new sample page to the show sample page

File: app/controllers/projects/samples_controller.rb [hunk 5]
```diff
@@ -50,7 +58,7 @@ def update
       respond_to do |format|
         if ::Samples::UpdateService.new(@sample, current_user, sample_params).execute
           flash[:success] = t('.success')
-          format.html { redirect_to namespace_project_sample_path(id: @sample.id) }
```
Comment (deepsidhu85) [inline app/controllers/projects/samples_controller.rb:53]: You can also revert this change and add a data: {turbo: false} to the form signature in `samples/edit.html.erb`. It will look like this:

`<%= form_with(model: @sample, url: namespace_project_sample_path(id: @sample.id), data: {turbo: false}, method: :patch) do |form| %>`

This will properly redirect from the edit sample page to the show sample page

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Just have a few more comments to get the redirect from the create/edit sample pages working correctly

File: app/models/sample.rb [hunk 6]
```diff
@@ -22,4 +22,18 @@ def self.ransackable_attributes(_auth_object = nil)
...
     %w[]
   end
+
+  def metadata_with_provenance
+    sample_metadata = []
+    metadata.each do |metadata_field, value|
+      provider = if metadata_provenance[metadata_field]['source'] == 'user'
+                   User.find(metadata_provenance[metadata_field]['id']).email
+                 else
+                   "Analysis #{metadata_provenance[metadata_field]['id']}"
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply app/models/sample.rb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

File: app/views/projects/samples/metadata/_table.html.erb (hunk 1 above)
Comment (ChrisHuynh333) [reply app/views/projects/samples/metadata/_table.html.erb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

File: app/views/projects/samples/metadata/_table.html.erb (hunk 2 above)
Comment (ChrisHuynh333) [reply app/views/projects/samples/metadata/_table.html.erb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

File: app/views/projects/samples/metadata/_table.html.erb (hunk 3 above)
Comment (ChrisHuynh333) [reply app/views/projects/samples/metadata/_table.html.erb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

File: app/models/sample.rb [hunk 7]
```diff
@@ -22,4 +22,18 @@ def self.ransackable_attributes(_auth_object = nil)
...
+                 end
+      sample_metadata << { metadata_field:, value:, provider:,
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply app/models/sample.rb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

File: app/controllers/projects/samples_controller.rb (hunk 4 above)
Comment (ChrisHuynh333) [reply app/controllers/projects/samples_controller.rb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

File: app/controllers/projects/samples_controller.rb (hunk 5 above)
Comment (ChrisHuynh333) [reply app/controllers/projects/samples_controller.rb]: Fixed in 0fb8fc1c66eb29d81ced91306361fadc7cb1d8b3

**Review by deepsidhu85 — APPROVED** [review]

**Review by joshsadam — APPROVED** [review]
This looks really good.
